src/extension/gogoduck/src/main/java/au/org/emii/gogoduck/worker/GoGoDuckModule.java

```diff
@@ -0,0 +1,212 @@
+        return ncksExtraParameters;
+    }
+
+    protected List<Attribute> getGlobalAttributesToUpdate(NetcdfFile nc) {
+        List<Attribute> newAttributeList = new ArrayList<Attribute>();
+
+        String title = profile;
+        try {
+            title = nc.findGlobalAttribute("title").getStringValue();
+
+            // Remove time slice from title ('something_a, something_b, 2013-11-20T03:30:00Z' -> 'something_a, something_b')
+            title = title.substring(0, title.lastIndexOf(","));
+        }
+        catch (Exception e) {
+            // Don't fail because of this bullshit :)
+            logger.warn("Could not find 'title' attribute in result file");
+        }
+
+        newAttributeList.add(new Attribute("title",
+                String.format("%s, %s, %s",
+                        title,
+                        subset.get("TIME").start,
+                        subset.get("TIME").end)));
+
+        newAttributeList.add(new Attribute("geospatial_lat_min", subset.get("LATITUDE").start));
+        newAttributeList.add(new Attribute("geospatial_lat_max", subset.get("LATITUDE").end));
+
+        newAttributeList.add(new Attribute("geospatial_lon_min", subset.get("LONGITUDE").start));
+        newAttributeList.add(new Attribute("geospatial_lon_max", subset.get("LONGITUDE").end));
+
+        newAttributeList.add(new Attribute("time_coverage_start", subset.get("TIME").start));
+        newAttributeList.add(new Attribute("time_coverage_end", subset.get("TIME").end));
+
+        return newAttributeList;
+    }
+
+    public final void updateMetadata(Path outputFile) {
+        try {
+            NetcdfFileWriter nc = NetcdfFileWriter.openExisting(outputFile.toAbsolutePath().toString());
+
+            nc.setRedefineMode(true);
+            for (Attribute newAttr : getGlobalAttributesToUpdate(nc.getNetcdfFile())) {
+                nc.addGroupAttribute(null, newAttr);
+            }
+            nc.setRedefineMode(false);
+            nc.close();
+        }
+        catch (IOException e) {
+            throw new GoGoDuckException(String.format("Failed updating metadata for file '%s': '%s'", outputFile, e.getMessage()));
+        }
+    }
+
+    public SubsetParameters getSubsetParameters() {
+        // Remove time parameter as we don't need to subset on it, we already
+        // have only files that are in the correct time range
+        SubsetParameters subsetParametersNoTime = new SubsetParameters(subset);
+        subsetParametersNoTime.remove("TIME");
+        return subsetParametersNoTime;
+    }
+
+    public static GoGoDuckModule newInstance(String profile, String geoserver, String subset, UserLog userLog) {
+        String thisPackage = GoGoDuckModule.class.getPackage().getName();
+        String classToInstantiate = String.format("GoGoDuckModule_%s", profile);
+
+        GoGoDuckModule module = null;
+        while (null == module && "" != classToInstantiate) {
```

Comment:
`"" != classToInstantiate ` will always evaluate to `true` in java.

Could be re-written as:

```java
while (null == module && !classToInstantiate.isEmpty())
```